include/eve/module/real/core/function/regular/simd/x86/convert_256.hpp

```diff
@@ -30,6 +30,14 @@ namespace eve::detail
     {
       return v0;
     }
+    else if constexpr( current_api >= avx2 && (N::value == 4)
+                       && std::is_integral_v<In> && (sizeof(In) == 8)
+                       && std::is_integral_v<Out> && (sizeof(Out) == 4) )
+    {
+      // u64 to u32
+      const auto p = _mm256_permutevar8x32_epi32(v0, _mm256_set_epi32(7, 5, 3, 1, 6, 4, 2, 0));
+      return _mm256_extractf128_si256(p, 0);
```

Comment:
Please use: `_mm256_castsi256_si128 `
I'd hope compiler optimises this but it's not harder.

Reply:
`_mm256_extractf128_si256` already returns a `__m128i` by design : https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_mm256_extractf128_si256&ig_expand=2989

Reply:
That's not what I meant.

You don't need extract. Extract has a latency 3. Cast does the same thing but with latency 0.

Llvm can optimize that but I don't know about MSVC.

Generally we prefer not to rely on optimizer if it doesn't cost us anything

Reply:
So put `_mm256_castsi256_si128` instead please

Reply:
Done in https://github.com/jfalcou/eve/pull/1049/commits/6a6887de53bb7b78be9159c4dbdf642d5e92de73